presto-raptor/src/main/java/com/facebook/presto/raptor/storage/OrcStorageManager.java

```diff
@@ -313,6 +316,48 @@ public ConnectorPageSource getPageSource(
         }
     }
 
+    Optional<BitSet> getRowsFromUuid(FileSystem fileSystem, Optional<UUID> deltaShardUuid)
+    {
+        if (!deltaShardUuid.isPresent()) {
+            return Optional.empty();
+        }
+        try (OrcDataSource dataSource = openShard(fileSystem, deltaShardUuid.get(), defaultReaderAttributes)) {
+            AggregatedMemoryContext systemMemoryUsage = newSimpleAggregatedMemoryContext();
+            OrcReader reader = new OrcReader(
+                    dataSource,
+                    ORC,
+                    defaultReaderAttributes.getMaxMergeDistance(),
+                    defaultReaderAttributes.getTinyStripeThreshold(),
+                    HUGE_MAX_READ_BLOCK_SIZE,
+                    orcFileTailSource);
+            if (reader.getFooter().getNumberOfRows() >= Integer.MAX_VALUE) {
```

Comment:
`checkArgument` will throw `RuntimeException` which will be classified below.

Reply:
Is it because `BitSet` can only handle int indexes? We will then need to make sure we always write as Integer in delta file, i.e. instead of read/write `BIGINT`, we read/write `INTEGER`.